providers/oidc.go

```diff
@@ -18,12 +24,19 @@ type OIDCProvider struct {
 	*ProviderData
 
 	SkipNonce bool
+
+	UseAssertionAuthentication bool
+	AssertionAuthJWTKey        *ecdsa.PrivateKey
+	AssertionAuthAlgorithm     string
+	AssertionAuthSigningMethod jwt.SigningMethod
+	AssertionAuthKeyId         string
+	AssertionAuthExpire        time.Duration
```

Comment:
I'm thinking about putting this into it's own sub struct and maybe move it one level higher. As it is not just part of the oidc specification but oauth2 in general. So we should keep in mind. That other providers might want to support it in the future as well 🤔

Something like a generic config inside the provider struct.

```
type Provider struct {
  ...
  AuthenticationConfig 🤔
}
```

Reply:
What do you think about that?